src/python/pants/backend/debian/target_types.py

```diff
@@ -13,14 +16,38 @@
 from pants.util.docutil import bin_name, doc_url
 
 
-class DebianControlFile(MultipleSourcesField):
+class DebianSources(MultipleSourcesField):
     required = True
-    expected_num_files = 1
     help = (
-        "Path to a Debian control file for the package to be produced.\n\n"
+        "Paths that will be included in the package to be produced."
+        "It is required to include a Debian control file.\n\n"
         "Paths are relative to the BUILD file's directory."
     )
 
+    def validate_resolved_files(self, files: Sequence[str]) -> None:
+        """Check that all files are coming from the same directory."""
+        if not files:
+            raise InvalidFieldException(
+                f"The `{self.alias}` field in target `{self.address}` must "
+                f"resolve to at least one file."
+            )
+
+        files_outside_dirs = [f for f in files if len(PurePath(f).parts) == 1]
+        if files_outside_dirs:
+            raise InvalidFieldException(
+                f"The `{self.alias}` field in target `{self.address}` must be paths to "
+                f"files in a single sources directory. Individual files "
+                f"were found: {files_outside_dirs}"
+            )
+
+        directory_prefixes = {PurePath(f).parts[0] for f in files}
+        if len(directory_prefixes) > 1:
+            raise InvalidFieldException(
+                f"The `{self.alias}` field in target `{self.address}` must be paths to "
+                f"files in a single sources directory. Multiple directories "
+                f"were found: {directory_prefixes}"
+            )
```

Comment:
Would be easy to write unit tests for these to avoid adding an additional integration test only to validate the `BUILD` file target declaration? The integration test take on average around 1 second.

Reply:
Yes, great idea! So you could add `target_types_test.py` with a test like this:

https://github.com/pantsbuild/pants/blob/7e97ef6dfd50a8eb0294b6ded5e67dbd7da921df/src/python/pants/engine/internals/graph_test.py#L1538-L1569

Reply:
Done!